src/com/mraof/minestuck/util/Echeladder.java

```diff
@@ -83,6 +87,8 @@ public void increaseProgress(int exp)
 					break increasment;
 				if(rung > prevRung + 1)
 					exp = (int) (exp/1.5);
+				EntityPlayer player = identifier.getPlayer();
+				player.world.playSound((EntityPlayer)null, player.posX, player.posY, player.posZ, MinestuckSoundHandler.soundUpcheladder, SoundCategory.AMBIENT, 1F, 1.0F);
```

Comment:
If you put it here, it will stack with each rung you get at the time. At least I think sound effects would stack. At least, to play the sound only once, it would be a good idea to put it further down in the if-statement checking that "rung != prevRung". I don't know if there'll be a moment when a player can get exp without being online, but just in case, you should probably check if "player" is null before playing the sound effect.
With another thing, it might be possible that the sound effect also plays for other people nearby. You should check that. And if they do, you could probably prevent that by instead playing the sound client-side in the PlayerDataPacket, around where the gain-rung messages are sent.